src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -852,4 +826,65 @@ private Response createUserObjectAndLogIn(final HttpServletRequest request, fina
                     "You cannot register with an Isaac email address.").toResponse();
         }
     }
+
+
+    /**
+     * Convert user-provided preference maps to UserPreference lists.
+     *     Contains Isaac classes which ideally should not be here . . .
+     *
+     * @param userPreferenceObject
+     *            - the user-provided preference object
+     * @param userId
+     *            - the userId of the user
+     * @return whether the preference is valid
+     */
+    private List<UserPreference> userPreferenceObjectToList(final Map<String, Map<String, Boolean>> userPreferenceObject, final long userId) {
+        List<UserPreference> userPreferences = Lists.newArrayList();
+        if (null == userPreferenceObject) {
+            return userPreferences;
+        }
+        // FIXME: This entire method is horrible, but required to sanitise what is stored in the database . . .
+        for (String preferenceType: userPreferenceObject.keySet()) {
+
+            // Check if the given preference type is one we support:
+            if (!EnumUtils.isValidEnum(IsaacUserPreferences.class, preferenceType)
+                    && !EnumUtils.isValidEnum(SegueUserPreferences.class, preferenceType)) {
+                log.warn("Unknown user preference type '" + preferenceType + "' provided. Skipping.");
```

Comment:
Arguably, if they provide us an invalid preference we should probably tell them and even throw an exception?

Reply:
Doing so breaks the whole atomicity thing: https://github.com/ucam-cl-dtg/isaac-app/issues/813 and really makes a mess when things go wrong. Failing silently isn't great, but at least it logs to the server logs, so we can ensure the front-end is doing the right thing.